rumqttd/src/server/broker.rs

```diff
@@ -297,12 +304,26 @@ impl Broker {
             })?;
         }
 
-        let console_link = ConsoleLink::new(self.config.console.clone(), self.router_tx.clone());
+        if let Some(console) = self.config.console.clone() {
+            let console_link = ConsoleLink::new(console, self.router_tx.clone());
+
+            let console_link = Arc::new(console_link);
+            let console_thread = thread::Builder::new().name("Console".to_string());
+            console_thread.spawn(move || {
+                let mut runtime = tokio::runtime::Builder::new_current_thread();
+                let runtime = runtime.enable_all().build().unwrap();
+                runtime.block_on(console::start(console_link));
+            })?;
+        }
 
-        let console_link = Arc::new(console_link);
-        let mut runtime = tokio::runtime::Builder::new_current_thread();
-        let runtime = runtime.enable_all().build().unwrap();
-        runtime.block_on(console::start(console_link));
+        // in ideal case, where server doesn't crash, join() will never resolve
+        // we still try to join threads so that we don't return from function
+        // unless everything crashes.
+        server_thread_handles.into_iter().for_each(|handle| {
+            // join() might panic in case the thread panics
+            // we just ignore it
+            let _ = handle.join();
+        });
```

Comment:
are panic messages still printed if anyone the threads panic?

asking since docs of join say it returns panic error message as `Err`.

if its not printed we should print them here.

Reply:
yes, the panic messages are printed. Tested by adding `panic!(..)` in `consume()` method of router so it can panic easily lol